src/main/java/com/bnkrll/service/SessionRepositorySimulator.java

```diff
@@ -0,0 +1,48 @@
+package com.bnkrll.service;
+
+import com.bnkrll.model.Session;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Slf4j
+@Repository
+public class SessionRepositorySimulator implements SessionRepository {
+
+    private final Map<String, Session> data;
+
+    public SessionRepositorySimulator() {
+        this.data = new HashMap<>();
+    }
+
+
+    @Override
+    public Session findById(String sessionID) {
+        return this.data.get(sessionID);
+    }
+
+    @Override
+    public void save(Session session) {
+        this.data.put(session.getSessionId(), session);
+        log.info("Session with id {} saved!", session.getSessionId());
+    }
+
+    @Override
+    public List<Session> getLastSessions(int numofSessions) {
```

Comment:
```suggestion
    public List<Session> getLastSessions(int numOfSessions) {
```

My IDE hinted that there was a spelling error here. 

Try installing a plugin called SonarLint and pay attention to those yellow and green underlines that the IDE shows you.